### ISSUE TITLE
Redesign the status report and available updates pages.

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#485]
As per suggestion from @Bojhan in #283 we should probably evaluate the effectiveness of the backdrop status report, and redesign as necessary.  

I've added the `audience - design` tag, if we have anyone with user interface or design experience, this would be a good place to lend a hand :)

![screen shot 2015-06-11 at 5 45 51 pm](https://cloud.githubusercontent.com/assets/397895/8120896/c167a06a-1061-11e5-897b-c2d50bc8fdd2.png)

---
latest PR by @jenlampton: https://github.com/backdrop/backdrop/pull/1836

**Comment by wesruv** [comment]
Can someone provide a list of statuses that are possible?

Is it:
* emergency/critical
* warning
* informational
* good

Missing anything?

Think we should borrow from the Recent Log messages design that we did a ways back that got merged in.

**Comment by quicksketch** [comment]
> Can someone provide a list of statuses that are possible?

That's it. From `[theme_status_report()](https://api.backdropcms.org/api/backdrop/core!modules!system!system.theme.inc/function/theme_status_report/1)`:

```
  $severities = array(
    REQUIREMENT_INFO => array(
      'title' => t('Info'),
      'class' => 'info',
    ),
    REQUIREMENT_OK => array(
      'title' => t('OK'),
      'class' => 'status',
    ),
    REQUIREMENT_WARNING => array(
      'title' => t('Warning'),
      'class' => 'warning',
    ),
    REQUIREMENT_ERROR => array(
      'title' => t('Error'),
      'class' => 'error',
    ),
  );
```

**Comment by wesruv** [comment]
Howz about something like this:

![image](https://cloud.githubusercontent.com/assets/5607236/14234464/5ae942d2-f9b1-11e5-96d9-9ad1a681f2cb.png)

The color coding of rows seemed like a bit much. I think a nice icon on the left can do the job better.

Wanted to pull out the short status message visually, and bring it closer to the status title. Think the 'mark' styling works pretty well, a little iffy on whether or not this is a valid use of the 'mark' style... thoughts?

**Comment by bd0bd** [comment]
Nice

**Comment by docwilmot** [comment]
yes

**Comment by jenlampton** [comment]
love it!

**Comment by klonos** [comment]
Yes, :+1: ...took a few seconds to get used to it, but it matches the rest of the admin UI better.

**Comment by Dinornis** [comment]
👍 ❤️ Great job @wesruv I like it, although slightly larger symbols inside the icons would be better.

**Comment by klonos** [comment]
Long-forgotten issue because of no assignee or label I guess. Getting this back into @wesruv's radar for when he has time 😉

**Comment by klonos** [comment]
Given the fact that we haven't settled on a specific design yet and also that @wesruv is not available lately, I thought that I might give this a try. I only aim to implement a few simple fixes without altering the current design that much. So, a few thoughts first...

1. [As it is currently implemented](https://github.com/backdrop/backdrop/blob/1.x/core/modules/system/css/system.admin.css#L141-L146), we have icons showing only for warnings and errors on that page. IIRC we've done this intentionally, in order to declutter the page and to take the emphasis away from the info and "OK" (status) rows. That is fine in general, but when there are only info and status rows on that page (basically when the site status is OK), then there is what seems to be an unnecessary padding on the left of the rows:

![backdrop-issue485-clean_status_report](https://cloud.githubusercontent.com/assets/2423362/23853395/7a2b99bc-0840-11e7-8b6c-215f3ea43b01.png)

...I initially thought that this would be easily fixed by just making some padding adjustments, but the horizontal "gap" is actually created because [we (unconditionally) still render the table cells that are to hold the row icons even if there are no icons to be rendered in them](https://github.com/backdrop/backdrop/blob/1.x/core/modules/system/system.theme.inc#L302). So, should we reintroduce the icons for info and status messages (simple css change), or should we adapt the logic (php) so that the entire column is not rendered when there are no icons in any of the rows?

2. Here's a screenshot of a series of test rows of all types (info/status/warning/error) and also in various combinations of a) simple rows, b) rows with a short description c) rows with long description text:

![backdrop-issue485-test_status_msgs_of_all_types](https://cloud.githubusercontent.com/assets/2423362/23878694/062d59a8-089d-11e7-85df-a15c304bfd69.png)

What's wrong here is that there is no distinction between the row title and the row text or the description which makes it hard to read this table. We also have a big distance between the row title and the text which makes it harder for the eye. The fact that there is no zebra striping going on makes this even worse.

This can be easily mitigated if we make some small changes like making the row title text bold and have the row titles occupy only as much width as the longest of them requires in order to be rendered without wrapping in a second line:

![backdrop-issue485-pr1802](https://cloud.githubusercontent.com/assets/2423362/23879288/318e87fe-08a0-11e7-8594-a305c8bb3dc6.png)

PR up for review with only the changes suggested in point 2 above. Pending decision on point 1.

**Comment by klonos** [comment]
...by the way, the PR is a quick fix to improve the situation even in a 1.6.x release. We should still keep pursuing a complete redesign of this page and also make it responsive in a next minor release.

Here's [what's in Drupal 8.3](https://www.drupal.org/node/665790) for reference:

No errors:
![d8_status_page-no_errors](https://cloud.githubusercontent.com/assets/2423362/23879759/52195c8a-08a3-11e7-9569-6ac124b9b2dd.png)

With errors:
![d8-status_page-with_errors](https://cloud.githubusercontent.com/assets/2423362/23879834/c57ab232-08a3-11e7-8915-03a7f352ee9e.png)

Mobile:
![d8-status_page-mobile](https://cloud.githubusercontent.com/assets/2423362/23879844/cf1817b2-08a3-11e7-84f1-18ea30cadd32.png)

**Comment by olafgrabienski** [comment]
@klonos I had a look at the PR and think it's a good start due to much better readability.

I'm not yet sure if I like the changes of the Status report page in Drupal 8.3, but it would really be good to make the page responsive as well. Couldn't we reach that goal simply by using `div` containers instead of a table?

**Comment by jenlampton** [comment]
I'm with @olafgrabienski, I'm not convinced the 8.3 redesign is a usability improvement - it definitely looks prettier, but IMHO would also be harder to use. The only thing that I think we may want to copy are the three boxes at the top for ERRORS / WARNINGS / CHECKED. 

Also about the icons, I don't think the icons were left out on purpose, it's likely it was an oversight. I, for one, would love to see an [i icon](http://fontawesome.io/icon/info/) for info and a [checkmark](http://fontawesome.io/icon/check/) for good.

**Comment by jlfranklin** [comment]
Moving the General System Information to its own page would prevent the errors list from falling below the fold, and put the more "information only" links in their own place.  My primary purpose for the Status Report page is to make sure nothing is broken or missing on the site.  I'm looking for red.

Do other modules that include / require JS libraries show up in the General System Information section or down below in the "Checked" section?  (Or in a fourth group?)

I haven't played with the new version in D8, but I expect those "details" links scroll you down to the appropriate section.

**Comment by laryn** [comment]
Design changes here could also be reflected on the "Available Updates" page (e.g. icons to the left, perhaps removing the colored background, etc.)

It would be great to be able to filter the status page at the top, the way [Module Filter](https://www.drupal.org/project/module_filter) allows on module / available updates page*:

![module_filter-3 0-update_status](https://cloud.githubusercontent.com/assets/871421/24117877/d64cabda-0d79-11e7-930b-01591e48f953.png)

_*Side note: it would also be great to have that extended filtering capability on the module / available updates page in Backdrop_

**Comment by klonos** [comment]
Yeah, I'm not sure about the D8 design either (especially the fancy logos/icons etc). I just put it there for reference. The good thing is that they have the resources to do usability tests, so there must be valid reasons for the changes.

What I like is that they've put all the server environment info (versions of core/apache/mysql/php etc.) together instead of having them spread all over the place. And to be honest, grouping things by severity level instead of alphabetically does make better sense.

As for mobile-friendly, you'll notice that in the current design what "consumes" the width of the page is the row description text. We could use fieldsets that are collapsed by default (perhaps have the critical errors be expanded).

**Comment by klonos** [comment]
>Also about the icons, I don't think the icons were left out on purpose, it's likely it was an oversight. I, for one, would love to see an i icon for info and a checkmark for good.

Yeah, I have already tried that @jenlampton, but having checkmarks everywhere looked kinda overwhelming ...and kinda unnecessary; especially in "all green" status:

![backdrop-issue485-icons_added_for_info_and_status-all_green](https://cloud.githubusercontent.com/assets/2423362/24140645/e995a2f4-0e74-11e7-8677-2cbdde3eacd4.png)

...looks a bit better with the background color removed from the rows (font colors retained):

![backdrop-issue485-icons_added_for_info_and_status-all_green-bg_color_removed_from_rows](https://cloud.githubusercontent.com/assets/2423362/24140717/1e3c4bfc-0e75-11e7-95ad-f249536177c4.png)

...perhaps if we added zebra stripping??:

![backdrop-issue485-icons_added_for_info_and_status-all_green-bg_color_removed_from_rows-zebra_stripping_added](https://cloud.githubusercontent.com/assets/2423362/24144880/0856a162-0e83-11e7-9c33-2bf177b9d28d.png)

**Comment by klonos** [comment]
...talking about zebra-stripping, the table in this page is not rendered through `theme_table()`, so we don't get even/odd classes. Seems that we are are doing custom table rendering in order to add the `.merge-down`/`.merge-up` classes that remove the border between the "combo" rows (the ones that have description text).

**Comment by klonos** [comment]
> Moving the General System Information to its own page would prevent the errors list from falling below the fold, and put the more "information only" links in their own place. My primary purpose for the Status Report page is to make sure nothing is broken or missing on the site. I'm looking for red.

Yeah, under "normal" circumstances, these will remain "info" rows, but if say you move your installation to another host that has an incompatible version of php, then that will become a warning or error. What would we do in such cases? ...duplicate the status on both pages?

> I haven't played with the new version in D8, but I expect those "details" links scroll you down to the appropriate section.

Yep, that's what they do. More useful on mobile and on narrower screens where the "General Info" section pushes the list of statuses further down.

**Comment by klonos** [comment]
> Design changes here could also be reflected on the "Available Updates" page (e.g. icons to the left, perhaps removing the colored background, etc.)

I agree. Care to open a separate issue for that.

> It would be great to be able to filter the status page at the top, the way [Module Filter](https://www.drupal.org/project/module_filter) allows on module / available updates page*:
> 
> _*Side note: it would also be great to have that extended filtering capability on the module / available updates page in Backdrop_

Way ahead of you @laryn 😄 (more than two years): #503 / #978 / #980

**Comment by klonos** [comment]
> > As for mobile-friendly...
> 
> Or we could add the appropriate classes to the table columns, like we do in the rest of core.

Given that hiding the icon column would only give us ~14px, we only have two main columns in that table that we could hide: requirement title and requirement value. If you hide either of them, then the table stops making sense.

**Comment by klonos** [comment]
...it ain't that bad:

![backdrop-issue485-icons_added_for_info_and_status-all_green-bg_color_removed_from_rows-zebra_stripping_added-tablet](https://cloud.githubusercontent.com/assets/2423362/24146651/b4a0e3fe-0e8a-11e7-8c14-7262e7b4939e.png)

...and it would still look good on even narrower screens with a small adjustment:

![backdrop-issue485-icons_added_for_info_and_status-all_green-bg_color_removed_from_rows-zebra_stripping_added-mobile](https://cloud.githubusercontent.com/assets/2423362/24146674/d44bfa40-0e8a-11e7-848f-bc0d9cd60cdb.png)

**Comment by klonos** [comment]
I have actually refactored the output of this page to be rendered via `theme_table()` and then have adjusted `theme_table()` so that `'rows'` accepts an additional `'merge'` attribute with possible values `'up'` or `'down'` (an API addition - not a change per se). So, now we can take advantage of the creation of "combo" rows in other places in the UI. Perhaps I'll extend it even further to allow `'merge'` to also accept `'both'` as a possible value, so that rows can be merged with the one before them as well as with the one after them simultaneously. Just no energy to file a PR now 😪 💤

**Comment by klonos** [comment]
...I'll go recharge a bit. In the meantime, I'd love some feedback on my ideas/suggestions and if you have any preferences like the colors to use for the blue on the info icon and the green on the checkmark icon. What you see on the screenshots is me being lazy about it and just using `royalblue` and `yellowgreen` respectively.

**Comment by klonos** [comment]
...also is the return of the icons good or bad?

**Comment by olafgrabienski** [comment]
Re Feedback: Are the last changes supposed to be in the PR?

**Comment by klonos** [comment]
Nope, you'll have to rely on the screenshots for now. What I've done in the current PR is to add the `@media` conditional so that the requirement titles are allowed to wrap when below 720px width. Once that happens, they share a 50% (roughly, because I'm not talking the icon column width into account) of the screen with the values column. This still keeps the requirement values close to their respective titles on wider screens for better readability (less distance for the eye to travel across the screen.)

**Comment by klonos** [comment]
...I've also tried reducing the `colspan` of the description text from 3 cells to 2 and then giving the icon cell a `rowspan` of 2 for "combo" rows with descriptions. Although that seemed more appropriate, it gave me a hard time trying to properly vertically align the icon when either the requirement title or the value (or both) were wrapping in more than on line (in smaller screen sizes).

**Comment by klonos** [comment]
Note to self: remember to join the yellow color or the "combo" zebra rows on mouse hover.

**Comment by olafgrabienski** [comment]
Thanks, @klonos . I was asking about the PR because to give a feedback it would be good to see the page also with some warnings or/and errors.

**Comment by klonos** [comment]
Yep. I will file a separate PR and also try to get it closer to @wesruv's [design](https://github.com/backdrop/backdrop-issues/issues/485#issuecomment-205040433). In order to get a warning preview, you can set the "Error messages to display" setting to anything else then the default "none" (did that already in the PR sandbox). As for errors, luckily the PR sandbox seems to be missing a db update of user.module, so anyone testing there please refrain from running the update.

...did I sai I nid to go to slip absut half an our ag 💤

**Comment by laryn** [comment]
@klonos I like, especially as a temporary design improvement while a more major overhaul may be thought through (_"...by the way, the PR is a quick fix to improve the situation even in a 1.6.x release. We should still keep pursuing a complete redesign of this page..."_) 

Nice work!

**Comment by olafgrabienski** [comment]
@klonos I like the direction as well!

One detail: If there are the icons we could use the same text color for all message types. Is probably better readable.

**Comment by klonos** [comment]
> One detail: If there are the icons we could use the same text color for all message types. Is probably better readable.

Yes, all but errors which is good if the stand out. As I said, I would like to bring the PR close to @wesruv's design where the requirement titles and descriptions have black/default font color, and the requirement values has a gray/green/orange/red "halo" for info/status/warning/error:

![](https://cloud.githubusercontent.com/assets/5607236/14234464/5ae942d2-f9b1-11e5-96d9-9ad1a681f2cb.png)

**Comment by klonos** [comment]
Alright, I think I did some pretty "unholy" things with `theme_table()` (for which I am open to suggestions on how could be otherwise achieved), but [it works for me](http://1802.backdrop.backdrop.qa.backdropcms.org/admin/reports/status) 🎉 (admin / 3MYZ3LD8) ...I have retained the table cells as they are currently (3 columns) + the improvements from the initial PR. Here's what it looks like:

![image](https://cloud.githubusercontent.com/assets/2423362/24249337/f696484c-1026-11e7-8ad1-023d7c79a4b3.png)

**Comment by klonos** [comment]
Note to reviewers: please do not freak out with the 20k+ failures on php53 and the 10k+ failures on php70. I'm sure they are random ...I wish 😆 ...I did say: "unholy things".

**Comment by olafgrabienski** [comment]
Hm, personally I don't like the "halos", the page looks quite cluttered with them. Maybe better to use them only on error and warning rows:

![screen-backdrop-status-report-modified](https://cloud.githubusercontent.com/assets/14188307/24250885/a781734c-0fd8-11e7-96d6-d3841afca434.png)

**Comment by klonos** [comment]
...and here is the same unholiness, but with the requirement title and value in the same table cell, wrapped in `<span>` tags (as per @wesruv's design):

![image](https://cloud.githubusercontent.com/assets/2423362/24251626/df45dcc2-102e-11e7-9345-d8cd83ef4d0a.png)

PR sandbox:

http://1820.backdrop.backdrop.qa.backdropcms.org/admin/reports/status
(admin / Acv9Pgfk)

**Comment by klonos** [comment]
Thanx for the feedback @olafgrabienski let's see what others say as well. I would like to discuss the implementation as well before I prepare a final PR. In the meantime 💤 and I'll catch you all in the dev meeting in a few hours.

**Comment by markabur** [comment]
Thanks for all the work on this!

Here's some feedback based on looking at the sandbox site and the screenshot above. Note that (a) I am *not* any sort of usability or accessibility expert and (b) I am red/green deficient, so I can give some input from a slightly-colorblind point of view.

Overall -- it's kind of hard to pick out the lines that need action.

The red icon color is great. It stands out the most, as it should. 

The red text actually feels the opposite, though. The bold black text stands out the most for me while the bold red text recedes visually.

The yellow icon color doesn't jump out as being very different from the green. I can tell that they are different, but I have to carefully scan down the page and look at each icon. The yellow lines should be second thing I look at (after the red lines), but for me they blend in with the green lines too much for it to work that way.

The icons would be significantly easier to differentiate if they were different shapes.

Some of the color combinations are difficult to read, for example (I didn't test everything):

- Blue link on medium gray background: http://webaim.org/resources/contrastchecker/?fcolor=0074bd&bcolor=cccccc 
- White icon on light green background: http://webaim.org/resources/contrastchecker/?fcolor=ffffff&bcolor=cfde56

If we're shoving everything over to the left then the alternating background colors become un-helpful, and could be dropped. 

The row-hover background color is not helpful, and think it probably reduces legibility since it reduces contrast.

This is probably a separate issue, but I have been using Drupal for years and I still don't really get the purpose of the "info" lines. Is there any reason not to display them the same as the "ok" lines? In other words, can we get away with three visual states instead of four? Why is Backdrop CMS "info" while Database system is "ok"?

**Comment by jlfranklin** [comment]
Regarding the "unholy things" and the 30k+ errors in the tests, all of the errors are the same error: 

> Exception Notice     theme.inc         2192 theme_table()                      
> Undefined variable: merge 

Line 2192 of theme.inc references `$merge`, but it has never been set to anything.  I haven't looked at the code, but I suspect you have a conditional block of some kind that sets `$merge`.  Initializing `$merge` before that block or at the top of the function will fix the tests.

**Comment by jenlampton** [comment]
> The icons would be significantly easier to differentiate if they were different shapes.

This is a good point. Maybe we should use the versions of the icons without circles around them to help differentiate?

info: http://fontawesome.io/icon/info/
check: http://fontawesome.io/icon/check/
exclamation: http://fontawesome.io/icon/exclamation/
times: http://fontawesome.io/icon/times/

> The red text actually feels the opposite, though. The bold black text stands out the most for me while the bold red text recedes visually.

I agree with this, and I don't have any color trouble (that I know of). Maybe we should make anything that's not a warning, not bold? And leave all the text black.

**Comment by docwilmot** [comment]
My preference for this is to float all alerts (non-green) to the top in a separate table styled like the last image in https://github.com/backdrop/backdrop-issues/issues/485#issuecomment-288046900 and have just a simple list for the green ones below.

**Comment by jenlampton** [comment]
I like this idea too, maybe we group everything together by status? In this
order: Red, yellow, green, gray.

**Comment by docwilmot** [comment]
Gray means 'information', which isn't necessarily "all's well". So I'd group it as an alert too.

See example. The terminology would need fixing, but the point is "cleared" messages dont need noticing, theyre just there for info, so no need for icons or decoration. Use the icons for the actual alerts.

![status report](https://cloud.githubusercontent.com/assets/1042283/24590657/c331cf5c-17b6-11e7-91a3-269fb16e69ab.png)

**Comment by jenlampton** [comment]
> Gray means 'information', which isn't necessarily "all's well". So I'd group it as an alert too.

Hm, okay, how about grouping them In this order: red, yellow, gray, green.  (This is the order D8 decided on too, looks like)

I was thinking we could also collapse the green section too, since those items would be read less of the time. Something like:

Errors Fieldset (collapsible / expanded) with table inside
Warnings Fieldset (collapsible / expanded) with table inside
Info Fieldset (collapsible / expanded) with table inside
Okay (Fieldset (collapsible / collapsed) with table inside

Hm, thinking more about that... with green at the bottom, I'm convinced it needs to be collapsed anymore... could go either way.

**Comment by docwilmot** [comment]
I dont think collapsible fieldsets work here: if you have alerts you want to see them.
Neither do I think separating the alert types is necessary: hopefully the vast majority of the time you'd have three empty fieldsets (unless you're being quite uncareful).

**Comment by jenlampton** [comment]
> if you have alerts you want to see them.

That's why they are expanded.  I was thinking making them collapsible would just make it easier to get past them if you wanted to get to the info section of the page, for example.

> hopefully the vast majority of the time you'd have three empty fieldsets

Well if they are empty, they wouldn't need be on the page at all, so you'd see only the gray, and then green things.

> Neither do I think separating the alert types is necessary

I'm not sure it's necessary either. I have lots of sites that have warnings (things like "some module is providing a weird token!" ) and that warning can be safely ignored.  I would want the errors that can't be ignored to come first, but they could be in the same table and/or fieldset (if we decide to use them).

**Comment by klonos** [comment]
> Note that (a) I am not any sort of usability or accessibility expert...

Neither am I. This shouldn't stop us from giving usability reviews ...even the amateur kind. I think we are entitled to that as users 😄 

> ...and (b) I am red/green deficient, so I can give some input from a slightly-colorblind point of view.

Valuable input 👍 ...perhaps this calls for a "for the visually impaired" version of the admin theme?

> The icons would be significantly easier to differentiate if they were different shapes.

Great idea! Note that the shapes are not part of the icons though. They are rendered via css.

> The row-hover background color is not helpful, and think it probably reduces legibility since it reduces contrast.

Yeah, this is the stock color that comes with tables and the zebra stripping. I had it on my to-do list, but didn't touch it in the PR. I think that based on the discussion during the dev meeting (or was it the design one? ...I forget), the next iteration will be based on divs instead of table rows/cells, so this will be a non-issue 😉 

>...I still don't really get the purpose of the "info" lines. Is there any reason not to display them the same as the "ok" lines? In other words, can we get away with three visual states instead of four?

I think that the intention there was to de-emphasize non-error/warning lines.

> Why is Backdrop CMS "info" while Database system is "ok"?

Beats me 😄 ...but things that can go red, should be green when ok IMO.

**Comment by klonos** [comment]
@jlfranklin thanx 👍 ...either rookie php dev mistake or omission because my PR was a PoC/test thing seeking for comments/approval before moving forward.

**Comment by klonos** [comment]
> > The icons would be significantly easier to differentiate if they were different shapes.
> 
> This is a good point. Maybe we should use the versions of the icons without circles around them to help differentiate?
> 
> info: http://fontawesome.io/icon/info/
> check: http://fontawesome.io/icon/check/
> exclamation: http://fontawesome.io/icon/exclamation/
> times: http://fontawesome.io/icon/times/

#364 anybody 😄 ...seriously though. There's so many places that we could use them, like menus including the admin bar (#1841), drop-buttons (#1974), the config page (#1178) etc.

> > The red text actually feels the opposite, though. The bold black text stands out the most for me while the bold red text recedes visually.
> 
> I agree with this, and I don't have any color trouble (that I know of). Maybe we should make anything that's not a warning, not bold? And leave all the text black.

I tried that, but then we have no distinction between row title and row status text (value). Makes it even worse (harder to scan the page for a specific entry) if we go with having values next to titles instead of aligning them vertically.

**Comment by klonos** [comment]
@jenlampton, @docwilmot with regards to using fieldsets, separating entries per severity level etc, I have been thinking #503 more and more lately. https://www.drupal.org/project/module_filter for example has an instant filter for the available updates page that allows showing all/update available/security update entries:

![module_filter-3 0-update_status](https://cloud.githubusercontent.com/assets/2423362/24615768/85f4fa5c-18d2-11e7-84ce-65e816646e15.png)

If we had something like that + an instant search text filter, then we'd keep the list in alphabetical order, stop making assumptions on what people expect to see when they come here and offer them a snappy way to choose the information they want to see.

**Comment by klonos** [comment]
...I see the same pattern being used in config pages on my android 7.0 and it is very convenient.

**Comment by jenlampton** [comment]
I do like the idea of having an instant search filter wherever it makes sense! 

However, I don't have a problem with making assumptions about what people are after when they visit certain pages. If we never assume anything and always leave it to the user to find it themselves, it's not going to be as good an experience as if we give them what they need without needing to look for it (or search for it).

If we build an experience that works with no effort for the 80% use case, and then provide a tool (like a filter box) for the other 20%, I think that's a best-case scenario!

**Comment by docwilmot** [comment]
TBH @jenlampton @klonos I really dont think this page needs to be anything but a simple list, as it is now. Its just a report, which in most cases will be all green, with no expectation of user interaction (or searching or hiding).

So to clarify my thinking and preference. Just order them by type with the errors at top. 

Maybe have the totally "green" ones in a separate table as I previously suggested just to de-emphasize them. @klonos using your android as an analogy, like the "About device" menu item. Just a plain ole list.

![tap-on-device-name-under-settings](https://cloud.githubusercontent.com/assets/1042283/24620341/ecd13bde-1862-11e7-87ae-59eca2a817a9.jpg)

**Comment by jenlampton** [comment]
@docwilmot that's a great suggestion and an easy win. We could push only the sort change (and maybe new colors & icons?) as a first step, then do a follow-up issue for a more extensive redesign - if we think it's warranted.

**Comment by markabur** [comment]
I would imagine that 80%(?) of the time the status page is visited is because a badge appears in the admin bar, so anything that helps with next-steps at that point is good. I see a "1" in a circle up there. I click it and see the status page. Hopefully I see one thing that needs to be done. 

*And* hopefully the page is designed in a way that doesn't distract me from that one thing. In the scenario above, if I need to run database updates, then I can -- and should -- ignore all of the other stuff on the page. In fact, that's what I do, but only because I already know that I don't care about things such as "Database system 4-byte UTF-8 support - Not enabled".

With that in mind I really like the idea of grouping, at least for the critical items, so they are bubbled up into their own to-do list.

> ...perhaps this calls for a "for the visually impaired" version of the admin theme?

Maybe not "for the visually impaired", but an admin theme that is specifically "high contrast"? It would be something to keep in mind if it's too difficult to hit those target WCAG contrast ratios. For me it's more just a matter of being busy, impatient, old, and grumpy. :-D I simply don't bother reading things that take extra time or effort -- too small, poor contrast, too many words, "click to reveal", "please wait while the next slide loads", etc..

**Comment by jenlampton** [comment]
> I simply don't bother reading things that take extra time or effort -- too small, poor contrast, too many words, "click to reveal", "please wait while the next slide loads", etc..

Weather we want to admit it or not, everyone behaves this way. If things are too hard to do, we don't do them - especially when it comes to reading things on websites!

**Comment by olafgrabienski** [comment]
I don't agree (or should I say "don't admit" ?-), no really, I like reading things on websites, guess we're just different on that. I however like the grouping idea as well.

**Comment by klonos** [comment]
I'm like @olafgrabienski myself. I even read all code in PR even if I don't understand it, trying to spot patterns and learn. I also go through the unchanged code in the 3 lines above and below the PR diffs and try to spot inconsistencies and typos (and get my thrills on each time I do ...even more now that I can file PRs myself). You know me... I'm the missed-a-spot guy 😄

**Comment by jenlampton** [comment]
@klonos and @olafgrabienski are the best!!!!

**Comment by wesruv** [comment]
@klonos I like what you have in the PR's, I prefer the non-2column variant.

I think the td's in the status report need a little more breathing room, I'm thinking this style:
```
table.system-status-report td {
    padding: 6px;
    vertical-align: middle;
}
```
Gets updated to be: `padding: 11px 6px`, and everything will feel less squished.

Otherwise I like it from a front end code/design standpoint! Great work!

**Comment by klonos** [comment]
I've added the second PR to the issue summary and updated both PRs with @wesruv's suggestion 👍 

What needs to be decided:

1. Keep the table or switch to divs?
2. Keep or ditch the yellow background on hover over each row (default core table behavior; needs to be added if we go with divs)? PS: I think yes because it improves readability.
3. Zebra stripes or not (either with table or with divs)? PS: again I think yes because it improves scannability.
4. Exact colors to be used:
  - blue or gray for info?
  - keep red for the error label/titles ([as originally designed](https://github.com/backdrop/backdrop-issues/issues/485#issuecomment-205040433) by @wesruv) or only for the value?

Do we have an "official" Seven/core color palette? ...I think we do, but does not include all colors required in this page.

**Comment by klonos** [comment]
...while we're at it, may I suggest a few other improvements that I either remember having been mentioned by others or are my personal opinions:

1. Merge the Backdrop version with the update status.
2. Merge the db system + db version + db updates status.
3. "GD library PNG support" and "GD library rotate and desaturate effects": are they ever different versions?
4. jQuery and jQuery UI: these were added by me in ??? as `'severity' => REQUIREMENT_OK,`, but I think they'd better be info instead. I mean, these are included with core; not environment-specific as db/php/web server which can fall below minimum system requirements.

**Comment by olafgrabienski** [comment]
@klonos Thanks for adding the 2nd PR. Of this two, I prefer the 3 columns variant which looks less cluttered.

Apart from that, I miss the grouping idea, see https://github.com/backdrop/backdrop-issues/issues/485#issuecomment-290994294 and below. I had the impression that most participants of this issue liked it.

**Comment by jenlampton** [comment]
@klonos I also prefer the 3 column variant as well.  And before I get to my other thoughts, a huge +1 to all the great feedback in this issue!  So much progress <3

Here are a few other thoughts:
* Sort by severity, then title?
* Swap out these icons for those without the circles.
* I'd also love to see the longer text only in the 3rd column. I think that will make it both more usable (easier to scan the column of titles on the left) and easier to read (fewer characters on a single line). * We could do the same trick as we use on the modules page to hide the more info when it's not needed. 
* I'm not sure how I feel about the background on the text, I feel it makes all the text that has that background harder to read.

I don't actually mind the use of a table here, especially if we can keep all the description text in a single row. I'm going to take a stab at a PR based on @klonos's first PR at the top. Something that looks like this: 
![screen shot 2017-04-13 at 5 31 56 pm](https://cloud.githubusercontent.com/assets/397895/25029157/7ef8c0fe-206f-11e7-95cb-13cb3f18d34f.png)

**Comment by jenlampton** [comment]
I think the mores should be open by default for errors and warnings, like this:
![screen shot 2017-04-13 at 5 50 13 pm](https://cloud.githubusercontent.com/assets/397895/25029398/c4ec9368-2071-11e7-9e83-3a8b244efe96.png)

**Comment by klonos** [comment]
> @klonos Thanks for adding the 2nd PR. Of this two, I prefer the 3 columns variant which looks less cluttered.

I think that I am leaning towards converting that list to use divs instead of it being a table. We can use a `<div class="row">` for each entry. Each row will have 4 `div`s inside it: one for the severity/icon, one for the label, one for the value and one for the description. No all rows will have content in the description div, but those who do will be wrapping it bellow the other 3 `div`s (for icon/label/value). This will allow us to have the following:

- Zebra striping without any "gimmicks" since the odd/even class will be applied to the parent `<div class="row">`
- Keep all info of each entry in a single row (with using a table, we had to use a second `<tr>` to hold the description). This helps with zebra as mentioned above, but also makes it easier to filter/sort the rows.
- We can have the label div be the same width as its content, thus bringing the vale div closer to it (simulating the 2 column design).
- People that like the old style, can create a subtheme of the Seven theme and make the label div fix width (simulating the 3 column design).

@olafgrabienski that last bit is for you and me that like the current 3-column design 😉

**Comment by klonos** [comment]
@jenlampton 👍 on the icons without the circles. The other ones looked so tiny. 👍 on sorting by severity, then title too.

Re the background on the values (we started referring to it as "halo", since it is not the background of entire row or table cell), yes, although it was included in the original design by @wesruv, some accessibility/contrast concerns made me have second thoughts too.

And finally, if we go with the 3 column design, then yes, we can keep the table and limit the description text to the 3rd column. This would be consistent with the modules page as already mentioned, so again 👍

**Comment by klonos** [comment]
Having the description align horizontally with the value makes so much more sense and makes the page more readable 👍

**Comment by klonos** [comment]
...minor comment/nit: when the value is a link and there is a description, the "more" link added by the description seems as if it is part of the value link. Also, we have some existing "more" links that jump to a separate page (php info for example), while the new "more" link will be just expanding/collapsing the description. A tiny bit of inconsistency on the "more" link behavior there.

**Comment by quicksketch** [comment]
I like the enhancements that @jenlampton added for hiding/showing in https://github.com/backdrop/backdrop/pull/1836, but the hiding/expanding didn't work in my testing (is there a JS file missing)?

Aesthetically I like the "halos" and am a bit sad to see them depart.

Regarding sorting... If we're sorting, we should put "Info" above "OK" items. And for some reason the entire list is reverse-alphabetical ("Web server" is first and "Access to update.php" is last).

**Comment by quicksketch** [comment]
Also, it would really help my reviewing if we can rally around one of these PRs instead of having 3 open ones. I don't have any strong preferences around DIVs vs. Tables.

Regarding the table stripping, I'd like to avoid changes to `theme_table()` if possible. I think we can accomplish what we want by setting `$row['no_striping']` on each row, and then manually setting `even` or `odd` classes on each row (the second row with a message matching the even/odd on the previous one).

**Comment by jenlampton** [comment]
> the hiding/expanding didn't work in my testing (is there a JS file missing)?

No, I didn't implement that since I wasn't sure it would pass muster. If people like it we can add that here, if we do I was also wondering, should that be abstracted out into it's own themable element?  We could make a little renderable that adds the js needed via `#attached` that could be used on the modules page, here, and for any other similar interface.

> And for some reason the entire list is reverse-alphabetical

This is due to a limitation in `backdrop_sort` where you can't choose ASC or DESC you can only choose SORT_STRING or SORT_NUMERIC. 

> Regarding sorting... If we're sorting, we should put "Info" above "OK" items. 

If we're not going to want these items sorted  by severity (with "Info" above "OK" items) then maybe we don't need to worry about the `backdrop_sort` problem. If we need to build our own order anyway, we'll still need two loops.

I'll take another stab at it.

**Comment by docwilmot** [comment]
what are halos?

**Comment by docwilmot** [comment]
never mind, found them. no, dont care much for halos, too busy looking for me.

**Comment by jenlampton** [comment]
Let's give this a shot: 
<img width="1492" alt="screen shot 2017-04-22 at 6 42 44 pm" src="https://cloud.githubusercontent.com/assets/397895/25309799/84f05d78-278b-11e7-8af3-a481896074cd.png">

Tablet: 
<img width="742" alt="screen shot 2017-04-22 at 6 42 23 pm" src="https://cloud.githubusercontent.com/assets/397895/25309800/8c9b9038-278b-11e7-85fd-9df5892c01ed.png">

Mobile: 
<img width="416" alt="screen shot 2017-04-22 at 6 42 14 pm" src="https://cloud.githubusercontent.com/assets/397895/25309801/92b09af4-278b-11e7-81fc-a9171e915a5d.png">

**Comment by olafgrabienski** [comment]
Had a look at the updated PR and like the current look and feel! Noted two issues though:

- The descriptions to be shown after clicking the "more" link are visible by default.
- After enabling the display of error messages (to get a yellow warning row), I saw the following notice: "Undefined index: severity in theme_status_report() (line 287 of /home/qa/github/backdrop/backdrop/pr/1836/core/modules/system/system.theme.inc)."

**Comment by jenlampton** [comment]
> The descriptions to be shown after clicking the "more" link are visible by default.

The descriptions for errors and warnings should be SHOWN by default, and the link should say "less".  Descriptions for info and okay should be hidden. Is this not what you are seeing?

> Undefined index: severity

Whoopsie, fixing now :)

**Comment by jenlampton** [comment]
> Undefined index: severity

Interesting, on that line it's simply separating out all the status messages by severity - and every status message should have a severity (I didn't change this part of the code). What modules are you running on this site? I wonder if one of them is incorrectly setting a requirement w/o a severity...

**Comment by olafgrabienski** [comment]
> The descriptions to be shown after clicking the "more" link are visible by default.

> oh yeah it's all broken... fixing now

Looks good at the moment!

**Comment by olafgrabienski** [comment]
Maybe slightly off-topic, but fits still in the title of this issue (*Redesign the status report*): I enabled and disabled the *Display of error messages* several times on the sandbox site of the PR for testing purposes. It's quite easy to go to the relevant settings page directly from the Status report since we introduced #2040. But when you save the *Display of error messages* setting, you don't return to the Status report automatically, and you have to find your way back.

To go back to the Status report, you can use the Admin menu (where exactly was I coming from?), or you just hit the browser *Back* button. Doing the latter, the Status report page won't be updated, showing the same status, which is confusing.

Let's add a redirection to the link so that the (clean) URL is `admin/config/development/logging?destination=admin/reports/status`!

**Comment by klonos** [comment]
> Aesthetically I like the background color "halos" and am a bit sad to see them depart.

Yeah, me too 😞 ...but the updated design does match the rest of the UI better.

Actual review of the PR:
👍 on consolidating things like the MySQL db system/version
👍 on grouping rows by severity and sorting alphabetically
👍 on the more/less expanding/collapsing
👍 on the bigger, circle-less severity icons
👍 on setting the 3rd column to low priority and hiding on mobile.
👍 on having the description on error rows expanded by default

A few notes/suggestions:
- Lets keep the classes consistent by naming them all `status-*`: we currently have `report-status`, `report-icon`, `status-title`, `mobile-summary`, `status-value`, `status-summary`, `status-description`. So, lets change the rest to `status-report`, `status-icon`, `status-summary-mobile`.
- Lets bring the status report, available updates and modules pages more in par:
    - we can consider wrapping the groups of severity in (collapsible?) fieldsets like the groups we have in the modules page and the core/modules/themes sections we have in the available updates page.
    - the `status-description` div should be `font-size: 0.9em; color: #444;` like the `admin-requirements` div in the modules page. Either that, or the other way around: change the font/color on the modules page to match.
- Operations (like "run cron", "run db update" and "rebuild permissions") should not be hidden inside the collapsed description divs; they should be buttons in a separate `status-operations` column. When switching to mobile, these operations are not available any more (because the 3rd column is hidden).
- While at it, can we please update the "There are security updates available for one or more of your modules or themes." text to also include layouts?

> I was also wondering: should that be abstracted out into it's own themable element?  We could make a little renderable that adds the javascript needed via `#attached` that could be used on the modules page, here, and for any other similar interface. (@todo)

BIG 👍 on that from me. I'll file a separate issue once I get the change 😉

**Comment by jenlampton** [comment]
> Let's keep the classes consistent by naming them all status-*

The reason for the new `report-` classes is that they are used in multiple reports so need to be general, not specific to the status report. 

> we can consider wrapping the groups of severity in (collapsible?) fieldsets

I thought of that too, but fieldsets are for forms only. These report pages are not forms :(

> Operations (like "run cron", "run db update" and "rebuild permissions") should not be hidden inside the collapsed description divs; they should be buttons in a separate status-operations column.

I do like the idea of making these actions into buttons! But I'm not sure about moving them into a separate column. This is a report, not one of our "manage" things interfaces (where there is an operation column). How about we put the buttons next to the title? ... Whatever we try, it would be good to see some rough mockups and choose a direction before we start onto the code.

> While at it, can we please update the "There are security updates available for one or more of your modules or themes." text to also include layouts?

Yes!

**Comment by quicksketch** [comment]
Latest PR has some more PHP notices showing up in tests:
```
exception Undefined variable: extra_section update.theme.inc:273
exception Undefined variable: base_section update.theme.inc:273
```

**Comment by jenlampton** [comment]
> Let's keep the classes consistent 

I changed the new class name to `status-summary-mobile`

> the status-description div should be font-size: 0.9em; color: #444;

done.

> can we please update the "There are security updates available for one or more of your modules or themes." text to also include layouts?

done.

> Latest PR has some more PHP notices showing up

fixed.

> Operations (like "run cron", "run db update" and "rebuild permissions")...

For the mobile display, I agree that we need to allow access to these links, but I don't think that we need to display the more/less description area. I tried to show it several times, with limited success. What I came up with is an API addition: for each requirement, a new element called "button" is allowed, it's an array with two keys: `tittle` for the text in the button, and `href` for where it should link. This will allow us to treat the button differently than the rest of the description, and pull it out for an easier display on mobile devices: 

<img width="546" alt="screen shot 2017-04-27 at 3 05 55 pm" src="https://cloud.githubusercontent.com/assets/397895/25506359/09e2c138-2b5b-11e7-931e-71d32f3855d9.png">

**Comment by jenlampton** [comment]
For buttons on errors on the status report, we could color them accordingly: 
<img width="552" alt="screen shot 2017-04-27 at 3 11 02 pm" src="https://cloud.githubusercontent.com/assets/397895/25506496/cc00053c-2b5b-11e7-8e89-9184fb2a0a31.png">

**Comment by markabur** [comment]
Looks good! Red button with white text would be too alarming, so it makes sense to go with red text on white bg with a border.

**Comment by jenlampton** [comment]
Okay, so @quicksketch doesn't like the fancy buttons (or the API addition) and made another recommendation that would require less rewriting of everything: move the "important" links into the status value - where the link for "PHP information" is located.

Desktop:
<img width="1211" alt="screen shot 2017-04-27 at 5 59 22 pm" src="https://cloud.githubusercontent.com/assets/397895/25510225/59bdc7da-2b73-11e7-8dd2-40514197977a.png">

Tablet:
<img width="747" alt="screen shot 2017-04-27 at 5 59 36 pm" src="https://cloud.githubusercontent.com/assets/397895/25510226/5bdbad0c-2b73-11e7-8590-529a82ebdd6d.png">

Mobile:
<img width="593" alt="screen shot 2017-04-27 at 5 59 50 pm" src="https://cloud.githubusercontent.com/assets/397895/25510228/5e5bf550-2b73-11e7-9e35-50f756b4aa9a.png">

**Comment by Dinornis** [comment]
I think the buttons would be okay with less padding.

**Comment by olafgrabienski** [comment]
The buttons are visually more clear but if moving the links is less problematic API-wise, it's a good idea to do so!

Looking at the mobile screen: there is no "important" link in the case of available security updates, which may confuse or scare users.

**Comment by klonos** [comment]
Yeah, I like the big, differently-styled buttons better too. But if the API addition is out of the question, oh well.

...can we at least try to make those links look like buttons in order to differentiate them from the more/less links?

**Comment by quicksketch** [comment]
> Looking at the mobile screen: there is no "important" link in the case of available security updates, which may confuse or scare users.

I pushed an update that adds a link with the text "Updates available".

> ...can we at least try to make those links look like buttons in order to differentiate them from the more/less links?

It seems I might be in the minority here regarding the buttons. How about if we make the API expansion and buttons a separate issue and call this one complete?

**Comment by jenlampton** [comment]
> How about if we make the API expansion and buttons a separate issue and call this one complete?
👍

**Comment by klonos** [comment]
...one minor nit I've noticed in the PR sandbox is that clicking the more/less link changes the width of the `status-value` column width.

**Comment by Dinornis** [comment]
Also, the `more` links disappear on screen widths with less than 734px in Safari and 720px in FireFox.
Not sure how I can trigger any status report errors or warnings.

**Comment by jenlampton** [comment]
> the more links disappear on screen widths with less than 734px

Yes, this is by design. We need that whole column to disappear on mobile.

> Not sure how I can trigger any status report errors or warnings.

Change the variable `$udpate_free_access` in settings.php to `TRUE` to get an error. For a warning you can install an older module or theme that has an update available (but not a security update, as that will also be an error)

**Comment by quicksketch** [comment]
> ...one minor nit I've noticed in the PR sandbox is that clicking the more/less link changes the width of the status-value column width.

Thanks @klonos fixed that minor issue up as well. Now the right column takes up as much space as allowed even if it doesn't need it, so when descriptions are opened it will not enlarge the column any further.

**Comment by quicksketch** [comment]
I've merged https://github.com/backdrop/backdrop/pull/1836 into 1.x for 1.7.0. Let's please create new issues for follow-ups (both large and small).
